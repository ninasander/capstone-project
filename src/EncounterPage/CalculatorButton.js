import React from 'react'
import styled from 'styled-components/macro'

export default function CalculatorButton({ operator, onClick }) {
  return (
    <ButtonStyled operator={operator} type="button" onClick={onClick}>
      {operator}
    </ButtonStyled>
  )
}

const operatorColors = {
  '+': 'green',
  '-': 'red',
}

const ButtonStyled = styled.button`
  font-size: 80%;
  border-style: none;
  width: fit-content;
  height: fit-content;
  text-align: center;
  background: ${(props) => operatorColors[props.operator]};
`
