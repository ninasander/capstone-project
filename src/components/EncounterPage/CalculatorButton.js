import React from 'react'
import styled from 'styled-components/macro'

export default function CalculatorButton({ operator, onClick }) {
  return (
    <ButtonStyled operator={operator} type="button" onClick={onClick}>
      {operator}
    </ButtonStyled>
  )
}

const operatorBgColors = {
  Healing: 'var(--light-green)',
  Damage: 'var(--light-red)',
}

const operatorColors = {
  Healing: 'var(--dark-green)',
  Damage: 'var(--dark-red)',
}

const ButtonStyled = styled.button`
  width: 65px;
  height: 30px;
  font-size: 80%;
  border-style: none;
  text-align: center;
  border-radius: 5px;
  padding: 2px 4px;
  background: ${(props) => operatorBgColors[props.operator]};
  color: ${(props) => operatorColors[props.operator]};
  border: 1px solid ${(props) => operatorColors[props.operator]};
`
