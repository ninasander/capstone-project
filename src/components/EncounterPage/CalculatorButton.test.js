import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import CalculatorButton from './CalculatorButton'
import userEvent from '@testing-library/user-event'
import { render } from '@testing-library/react'

describe('CalculatorButton', () => {
  it('renders addButton correctly', () => {
    const tree = renderer.create(<CalculatorButton operator="+" />)
    expect(tree).toMatchSnapshot()
  })
  it('renders subtractButton correctly', () => {
    const tree = renderer.create(<CalculatorButton operator="-" />)
    expect(tree).toMatchSnapshot()
  })
  it('calls onClick when button is clicked', () => {
    const onClickMock = jest.fn()

    const { getByText } = render(
      <CalculatorButton onClick={onClickMock} operator="+" />
    )
    const button = getByText('+')
    userEvent.click(button)

    expect(onClickMock).toHaveBeenCalled()
  })
})
