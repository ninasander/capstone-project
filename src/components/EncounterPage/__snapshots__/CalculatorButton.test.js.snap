// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorButton renders addButton correctly 1`] = `
.c0 {
  font-size: 80%;
  border-style: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  text-align: center;
  background: green;
}

<button
  className="c0"
  operator="+"
  type="button"
>
  +
</button>
`;

exports[`CalculatorButton renders subtractButton correctly 1`] = `
.c0 {
  font-size: 80%;
  border-style: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  text-align: center;
  background: red;
}

<button
  className="c0"
  operator="-"
  type="button"
>
  -
</button>
`;
